### ISSUE TITLE
what is difference between --javascript-delay and --no-stop-slow-scripts 

### ISSUE THREAD

**Issue by VinayParasrampuria** [wkhtmltopdf/wkhtmltopdf#2091]
Hello Sir ,

I am using wkhtmltopdf 0.12.2Dev version .I have used --no-stop-slow-scripts  to ensure all the scripts get finished before generating PDF . My question is in this situation what is the use of --javascript-delay.Do I need to specify --javascript-delay if there is   --javascript-delay mentioned in HTML page .

Regards
Vinay

**Comment by KantarBruceAdams** [comment]
In fairness to this user the documentation is quite slim. The text in https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4174 lists what the options do but does not say when or why you would use either of them.

**Comment by joshuapinter** [comment]
For what it is worth, we were generating very large PDFs with a lot of javascript that was required to run. The javascript was getting terminated with the message `Warning: A slow script was stopped`. 

I was initially going to use a `--javascript-delay` and was trying to figure out what value to use and how to make it dynamic based on page size, etc. But, instead, stumbled upon `--no-stop-slow-scripts` and that ended up working perfectly. 

In short, try using just `--no-stop-slow-scripts` first and see if it resolves your issue. If not, dig deeper.
